server/__tests__/integration/axios-cache-interceptor.test.js

```diff
@@ -0,0 +1,57 @@
+require('dotenv').config()
+const Axios = require('axios')
+const { setupCache } = require('axios-cache-interceptor')
+
+// 1. we test that the configuration we set is valid and available to check
+describe('test Axios configuration', () => {
+  test('tests argument composition', () => {
+    const axios = Axios.create()
+    const withAxios = setupCache(axios)
+    // we expect axios to be properly configured with setupCache
+    expect(withAxios).not.toBeUndefined()
+
+    // we expect the cache to be properly configured and contain our properties after initialization
+    const withConfig = setupCache(axios, { ttl: 1234 })
+    expect(withConfig).not.toBeUndefined()
+    expect(withConfig.defaults.cache.ttl).toBe(1234)
+  })
+})
+
+// 2. we test that the cache is working as expected with concurrent requests
+describe('test Axios cache with concurrent requests', () => {
+  test('tests cache', async () => {
+    const axios = Axios.create()
+    const cache_interceptor = setupCache(axios)
+
+    // we make two concurrent requests to the same endpoint
+    const req1 = cache_interceptor.get('https://api.publicapis.org/entries')
+    const req2 = cache_interceptor.get('https://api.publicapis.org/entries/')
+    const [res1, res2] = await Promise.all([req1, req2])
+
+    // assertions: we expect the second response to be cached
+    expect(res1.cached).toBe(false)
+    expect(res2.cached).toBe(true)
+  })
+})
+
+// 3. we test that the cache is invalidated when the ttl expires
+describe('test Axios cache invalidation option', () => {
+  test('tests cache invalidation', async () => {
+    // we set the ttl to 10 seconds
+    const axios = Axios.create()
+    const cache_interceptor = setupCache(axios, { ttl: 10000 })
+    // we make two concurrent requests
+    const req1 = cache_interceptor.get('https://api.publicapis.org/entries')
+    const req2 = cache_interceptor.get('https://api.publicapis.org/entries/')
+    const [res1, res2] = await Promise.all([req1, req2])
```

Comment:
Same problem as above - order of promise fulfillment is not guaranteed.  For this test, though, there is no need to make multiple requests again.  The purpose of this test is to make sure that the cached entry expires, so you can make a single request, wait for the ttl, and then verify that it is no longer cached.

Reply:
Since a previous method already testes if the caching works properly, I agree that I should only focus on the purpose of this test. Before, I also wanted to assert that all responses were being cached, prior to invalidating data, to make it more obvious. I have updated the file.